packages/contracts/src.ts/index.ts

```diff
@@ -1177,46 +1178,33 @@ export class ContractFactory {
         defineReadOnly(this, "signer", signer || null);
     }
 
-    // @TODO: Future; rename to populateTransaction?
-    getDeployTransaction(...args: Array<any>): TransactionRequest {
-        let tx: TransactionRequest = { };
+    getDeployTransactions(...args: Array<any>): Array<TransactionRequest> {
+        let chunks = splitInChunks(Buffer.from(this.bytecode).toString(), 4096);
 
-        // If we have 1 additional argument, we allow transaction overrides
-        if (args.length === this.interface.deploy.inputs.length + 1 && typeof(args[args.length - 1]) === "object") {
-            tx = shallowCopy(args.pop());
-            for (const key in tx) {
-                if (!allowedTransactionKeys[key]) {
-                    throw new Error("unknown transaction override " + key);
-                }
```

Comment:
We should leave this check. The reason why it exists is that the user can provide an additional argument `transaction overrides` as part of **every** transaction. 

The overrides that can be passed are:
- `value`
- `gasLimit`

![image](https://user-images.githubusercontent.com/12232750/151794320-434a7b9d-6e78-4cf8-a7ed-56901ed572b0.png)